src/transformers/utils/fx.py

```diff
@@ -121,107 +122,311 @@ def _generate_supported_model_classes(
 )
 
 
+def embedding_override(self, input):
+    return torch.empty(*input.shape, self.weight.shape[-1], device="meta")
+
+
+def torch_nn_layernorm_override(self, input):
+    return input
+
+
+def torch_nn_linear_override(self, input):
+    return torch.empty(input.shape[:-1] + (self.out_features,), device="meta")
+
+
+def torch_relu_override(x):
+    return x
+
+
+def torch_nn_relu_override(self, x):
+    return x
+
+
+def torch_nn_functional_relu_override(x, inplace=False):
+    assert not inplace, "dont support inplace functional.relu for metatensor analysis"
```

Comment:
I don't think that it's an issue here, because all the asserts are under a `try` block, if one of them raises an exception, it is handled later in the function and a [proper message is being raised](https://github.com/michaelbenayoun/transformers/blob/fx_with_meta/src/transformers/utils/fx.py#L589). Wdty?

Reply:
Asserts can be deactivated when executing code, so your `try` block will always pass. That's why we use Python errors everywhere else in the codebase.

Reply:
Alright, gotcha, I replaced the assertions by explicit exceptions.